include/arcade/Color.hpp

```diff
@@ -0,0 +1,100 @@
+/*
+** EPITECH PROJECT, 2022
+** Arcade
+** File description:
+** Color
+*/
+
+/// @file
+///
+/// The Color data structure.
+
+#ifndef ARCADE_COLOR_HPP_
+#define ARCADE_COLOR_HPP_
+
+#include <cstddef>
+#include <cstdint>
+
+namespace arcade
+{
+    /// 8bits colors used as backup in IGameObject interface (setForeground, setBackground).
+    enum class DefaultColor { Black, White, Transparent, Red, Green, Blue, Yellow, Magenta, Cyan };
+
+    /// 32bits ARGB Color.
+    class Color {
+      public:
+        /// Default constructor. Create a black color (all components set to 0).
+        constexpr inline Color() : a(std::byte(0)), r(std::byte(0)), g(std::byte(0)), b(std::byte(0))
+        {
+        }
+        /// Construct a color from an 32bits ARGB value.
+        ///
+        /// @param color 32bits ARGB color.
+        constexpr inline Color(uint32_t color)
```

Comment:
Prefer the `Color(uint_32_t): a(...), b(...), ... {}` notation, VSCode is showing me a compilation error here